### ISSUE TITLE
certbot-dns-godaddy==2.9.0

### ISSUE THREAD

**Issue by nicandris** [NginxProxyManager/nginx-proxy-manager#3592]
Hey, just updated to the latest docker img and i get the following error

```**[2/28/2024] [5:46:24 PM] [Certbot  ] › ▶  start     Installing godaddy...
[2/28/2024] [5:46:24 PM] [Global   ] › ⬤  debug     CMD: . /opt/certbot/bin/activate && pip install --no-cache-dir  certbot-dns-godaddy==$(certbot --version | grep -Eo '[0-9](\.[0-9]+)+')  && deactivate
[2/28/2024] [5:46:25 PM] [Certbot  ] › ✖  error     ERROR: Could not find a version that satisfies the requirement certbot-dns-godaddy==2.9.0 (from versions: 0.1.0, 0.1.1, 0.2.0, 0.2.1, 0.2.2, 0.2.3, 0.2.4, 2.6.0, 2.7.4, 2.8.0)
ERROR: No matching distribution found for certbot-dns-godaddy==2.9.0**
```

I don't see such version in [pypi](https://pypi.org/project/certbot-dns-godaddy/#history)
![image](https://github.com/NginxProxyManager/nginx-proxy-manager/assets/284798/e222c571-5041-4a71-b1b1-b35986fcbb63)

**Comment by nicandris** [comment]
for anyone else encountering this, for now hardcode the version 2.8.0 in `global/certbot-dns-plugins.json`

```        "godaddy": {
                "name": "GoDaddy",
                "package_name": "certbot-dns-godaddy",
                "version": "~=2.8.0",
                "dependencies": "",
                "credentials": "dns_godaddy_secret = 0123456789abcdef0123456789abcdef01234567\ndns_godaddy_key = abcdef0123456789abcdef01234567abcdef0123",
                "full_plugin_name": "dns-godaddy"
        },
```

**Comment by innosflew** [comment]
> for anyone else encountering this, for now hardcode the version 2.8.0 in `global/certbot-dns-plugins.json`
> 
> ```
>                 "name": "GoDaddy",
>                 "package_name": "certbot-dns-godaddy",
>                 "version": "~=2.8.0",
>                 "dependencies": "",
>                 "credentials": "dns_godaddy_secret = 0123456789abcdef0123456789abcdef01234567\ndns_godaddy_key = abcdef0123456789abcdef01234567abcdef0123",
>                 "full_plugin_name": "dns-godaddy"
>         },
> ```

Yeah, I also have this issue. Although, I'm not sure how to modify or even find `global/certbot-dns-plugins.json` when it comes to docker.

**Comment by ukewea** [comment]
On my Raspberry Pi 4 (arm64) using image with [2.11.1](https://hub.docker.com/layers/jc21/nginx-proxy-manager/2.11.1/images/sha256-a6231ef7f3b506fa98cf579a9bad77e46c401e0f90f875163e7ee414ec97ab2d?context=explore) tag, I have to change to `"version": "~=2.6.0",` or I would encounter the `Failed to renew certificate npm-2 with error: Error determining zone identifier for __domain__: 404 Client Error: Not Found for url: https://api.godaddy.com/v1/domains/__domain__` error when renewing certificate.

**Comment by TheRealVira** [comment]
Fixed it by installing a specific certbot version:

`pip3 install certbot==2.8`

This works, because it sets all dependency plugin versions to `2.8` as declared in `global/certbot-dns-plugins.json`:

```
...
        "godaddy": {
                "name": "GoDaddy",
                "package_name": "certbot-dns-godaddy",
                "version": "=={{certbot-version}}",
...
        }
```

The issue with certbot at version `2.9` is that there is currently no `certbot-dns-godaddy v2.8`: https://github.com/miigotu/certbot-dns-godaddy
![image](https://github.com/NginxProxyManager/nginx-proxy-manager/assets/10862533/66d8c960-1474-4740-88b2-cb5345d1cd91)

**Comment by shayaknyc** [comment]
How would one go about getting this fixed in the docker container?

**Comment by TheRealVira** [comment]
> How would one go about getting this fixed in the docker container?

https://stackoverflow.com/questions/30172605/how-do-i-get-into-a-docker-containers-shell#30173220

**Comment by shayaknyc** [comment]
> > How would one go about getting this fixed in the docker container?
> 
> https://stackoverflow.com/questions/30172605/how-do-i-get-into-a-docker-containers-shell#30173220

I know how to get INSIDE the container and execute things inside of it, but once you stop, reboot, etc, all changes are lost, so how do we make this change permanent?

**Comment by nicandris** [comment]
> How would one go about getting this fixed in the docker container?

copy the [file ](https://github.com/NginxProxyManager/nginx-proxy-manager/blob/develop/global/certbot-dns-plugins.json)from this repo, place it somewhere on your machine, do the change, and [mount it in docker](https://stackoverflow.com/questions/42248198/how-to-mount-a-single-file-in-a-volume)

it should be mounted to `/app/global/certbot-dns-plugins.json`

**Comment by snorkrat** [comment]
> > How would one go about getting this fixed in the docker container?
> 
> copy the [file ](https://github.com/NginxProxyManager/nginx-proxy-manager/blob/develop/global/certbot-dns-plugins.json)from this repo, place it somewhere on your machine, do the change, and [mount it in docker](https://stackoverflow.com/questions/42248198/how-to-mount-a-single-file-in-a-volume)
> 
> it should be mounted to `/app/global/certbot-dns-plugins.json`

Thanks, that fixed it for me!

**Comment by shayaknyc** [comment]
Me too!

**Comment by wolviex** [comment]
I created a pull request to pin the certbos-dns-godaddy version at the latest, v2.8.0.
It introduced some other issues for me, namely an issue with certbot/plugins/dns_common_lexicon.py regarding the recent update about delegated subdomains.

Try this PR at your own risk, but I'd appreciate the feedback if anyone else can test and confirm:
Docker Image for build 1 is available on [DockerHub](https://cloud.docker.com/repository/docker/jc21/nginx-proxy-manager) as `jc21/nginx-proxy-manager:github-pr-3639`

**Comment by CameronMunroe** [comment]
In my initial testing it seems I have fallen onto issues with subdomains with wildcards possibly mentioned by @wolviex.

`*.sub.example.com` for example will route the API request to `sub.example.com` instead of `example.com` on the GoDaddy API. Causing a 404 error. 

Unsure if this issue is an upstream certbot change or NPM.

**Comment by Red3nzo** [comment]
Currently running into this error after downgrading to `certbot==2.8`.
```console
Error determining zone identifier for <DOMAIN>: 401 Client Error: Unauthorized for url: https://api.godaddy.com/v1/domains/<DOMAIN>.
Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /tmp/letsencrypt-log/letsencrypt.log or re-run Certbot with -v for more details.
```

Does anyone have any suggestions? Would the `jc21/nginx-proxy-manager:github-pr-3639`  image pull fix this?

**Comment by wolviex** [comment]
No, `jc21/nginx-proxy-manager:github-pr-3639` would only remove the need for manually downgrading Certbot to 2.8.
I believe your issue is the same as I noted above regarding subdomain delegation.
I believe this was introduced in dns-lexicon 3.17 here: https://github.com/certbot/certbot/pull/9821 and I suspect that [certbot-dns-godaddy](https://github.com/miigotu/certbot-dns-godaddy) needs to accommodate for it. I'm looking at contributing to that repo anyways, so there may be some further updates to this issue.
### **TL;DR The `Error determining zone identifier` is not an NPM issue**


**HERE BE DRAGONS**
I haven't had a chance to dig in yet, but the workaround for me was to remove/comment out the line:
`204: 'delegated': domain,` in 
`/opt/certbot/lib/python3.11/site-packages/certbot/plugins/dns_common_lexicon.py`
This is a pain in the ass, and I never recommend changing site-package files.

**Comment by Red3nzo** [comment]
> remove/comment out the line: 204: 'delegated': domain,
I'm not familiar with python, I edited the site-package file & rerun the SSL Certificate creation & got the same `Error determining zone identifier`. 

Do I have to recompile the certbot package @wolviex?

**Comment by wolviex** [comment]
> Do I have to recompile the certbot package @wolviex?

You may need to restart the container

**Comment by Red3nzo** [comment]
Container was restarted & the problem still persists, line of code is commented out as well.

**Comment by nicandris** [comment]
solved by https://github.com/NginxProxyManager/nginx-proxy-manager/releases/tag/v2.11.2
